costmap_converter/src/costmap_converter_node.cpp

```diff
@@ -119,10 +119,9 @@ class CostmapStandaloneConversion : public rclcpp::Node {
 
     if (converter_) {
       converter_->setOdomTopic(odom_topic);
-      converter_->initialize(
-          std::make_shared<rclcpp::Node>("intra_node", "costmap_converter"));
+      converter_->initialize(shared_from_this());
       converter_->startWorker(std::make_shared<rclcpp::Rate>(5),
-                              costmap_ros_->getCostmap(), true);
+                              costmap_ros_->getCostmap(), false);
```

Comment:
That's the main reason for high CPU usage. This makes [this function](https://github.com/logivations/costmap_converter/blob/fix-high-CPU/costmap_converter/include/costmap_converter/costmap_converter_interface.h#L243) run which for some reason is really intense.

Reply:
possibly this while loop just spins too much, adding a Rate might also fix it.

If everything works without spinning at all, then we can also simply not do it. I would be worried that e.g. some subscribers stop working then